### PR TITLE
Display Nexus Service Versions

### PR DESCRIPTION
**Pull request by dhaneshnm**
Fetch and Display Nexus Service versions on information menu in the
Header.
implements BlueBrain/nexus#773

### CODE REVIEW

File: src/client/index.tsx [hunk 1]
```diff
@@ -69,9 +69,11 @@ const nexus = createNexusClient({
   uri: preloadedState.config.apiEndpoint,
   links: [setToken],
 });
+
```
Comment (dhaneshnm) [inline src/client/index.tsx]: I will remove these extra spaces in next commit.

File: src/shared/components/Header/index.tsx [hunk 2]
```diff
@@ -2,7 +2,9 @@ import * as React from 'react';
 import { Menu, Dropdown, Icon, Button, Popover } from 'antd';
 import './Header.less';
 import Copy from '../Copy';
-
+import { NexusClient } from '@bbp/nexus-sdk';
```
Comment (julienmachon) [inline src/shared/components/Header/index.tsx]: You cannot make use of the SDK (or any other side effect libraries) in a component. As a global strategy, this is why we have `components` and `containers`

File: src/shared/components/Header/index.tsx [hunk 3]
```diff
@@ -11,6 +13,32 @@ interface InformationContentProps {
   githubIssueURL: string;
 }
 
+const VersionInfo = () => {
+  const preloadedState: RootState = (window as any).__PRELOADED_STATE__;
```
Comment (julienmachon) [inline src/shared/components/Header/index.tsx]: You should not access the `__PRELOADED_STATE__` directly in a component. In fact you should never do this anywhere apart from when you bootstrap your app and you want to dehydrate the global state with the server-side computed state. 

If you need to connect part of your state to a component, use the `react-redux` library and the `connect` higher order function, and finally you need to make sure you do this in a `container` and not a `component`

**Comment by dhaneshnm** [comment]
Screenshot_2019-10-29 A knowledge graph for data-driven science

File: src/shared/components/Header/__tests__/Header.test.tsx [hunk 4]
```diff
@@ -1,17 +1,26 @@
 import * as React from 'react';
 import { shallow, mount } from 'enzyme';
-import Header from '..';
+import Header, {  serviceVersions } from '..';
+
+const versions:serviceVersions = {
```
Comment (julienmachon) [inline src/shared/components/Header/__tests__/Header.test.tsx]: there are not spaces between your variable and your type. Are you running prettier?

File: src/shared/components/Header/index.tsx [hunk 5]
```diff
@@ -9,8 +9,30 @@ const epflLogo = require('../../EPFL-logo.svg');
 interface InformationContentProps {
   version: string;
   githubIssueURL: string;
+  serviceVersions?: serviceVersions;
 }
 
+export  type serviceVersions = {
```
Comment (julienmachon) [inline src/shared/components/Header/index.tsx]: extra space between `export` and `type`. Also by convention, our types always start with a capital letter
```suggestion
export type ServiceVersions = {
```

File: src/shared/layouts/MainLayout.tsx [hunk 6]
```diff
@@ -37,13 +41,21 @@ const MainLayout: React.FunctionComponent<MainLayoutProps> = ({
...
+  apiEndpoint
 }) => {
   const handleLogout = (e: React.SyntheticEvent) => {
     e.preventDefault();
     localStorage.removeItem('nexus__state');
     userManager && userManager.signoutRedirect();
   };
 
+  
+  const apiBase = new URL(apiEndpoint);
```
Comment (julienmachon) [inline src/shared/layouts/MainLayout.tsx]: clever! :bulb:

File: src/shared/components/Header/__tests__/Header.test.tsx (hunk 4 above)
Comment (dhaneshnm) [reply src/shared/components/Header/__tests__/Header.test.tsx]: I have updated the code.

**Pull request merged**